superagi/models/agent_execution.py

```diff
@@ -164,4 +164,61 @@ def assign_next_step_id(cls, session, agent_execution_id: int, next_step_id: int
         if next_step.action_type == "ITERATION_WORKFLOW":
             trigger_step = IterationWorkflow.fetch_trigger_step_id(session, next_step.action_reference_id)
             agent_execution.iteration_workflow_step_id = trigger_step.id
-        session.commit()
\ No newline at end of file
+        session.commit()
+
+    @classmethod
+    def get_execution_from_agent_id_with_status(cls,session,agent_id,status_filter):
+        db_agent_execution = session.query(AgentExecution).filter(AgentExecution.agent_id==agent_id,AgentExecution.status==status_filter).first()
+        return db_agent_execution
+    
+    @classmethod
+    def get_updated_execution_config_obj(cls,agent_execution):
```

Comment:
move it to controller

Reply:
removed from model and moved to controller